src/components/LeaderboardAction.vue

```diff
@@ -0,0 +1,52 @@
+<template>
+  <div class="action-pill">
+    <BaseLink :to="{ name: path }" v-if="path">
+      <span class="mr-1">{{ text }} </span><BaseIcon :name="icon" />
+    </BaseLink>
+
+    <ShareButton :password="leaderboard.password" v-else />
+  </div>
+</template>
+
+<script>
+import ShareButton from '@/components/ShareButton'
+export default {
+  components: { ShareButton },
+
+  props: {
+    leaderboard: {
+      type: Object,
+      default: null,
+      required: false,
+    },
+    text: {
+      type: String,
+      default: null,
+      required: false,
+    },
+    icon: {
+      type: String,
+      default: null,
+      required: false,
+    },
+    path: {
+      type: String,
+      default: null,
+      required: false,
+    },
+  },
+}
+</script>
+
+<style lang="scss" scoped>
+@import '@/styles';
+.action-pill {
+  background-color: $light-gray;
+  color: $purple;
+  padding: $spacer / 2 $spacer * 2;
+  flex: 1;
+  border-radius: $spacer * 2;
+  margin: $spacer $spacer / 2;
+  text-align: center;
+}
+</style>
```

Comment:
No need to change for now, but whenever you create a scoped style, it creates a unique ID for those rules. So if you `@import` the styles in it, you're actually creating a duplicate of all the `styles` rules, just for that component. If we do this for too many components, it will add up and it might affect performance due to a large css.

Not a priority though, and eventually, I'll refactor this using Tailwind which uses shared classes, so let's keep it like this for now as long as it works.